src/Exception/ContainerExceptionInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Dhii\Data\Container\Exception;
+
+use Dhii\Data\Container\ContainerAwareInterface;
+
+/**
+ * An exception that occurs in relation to a container.
+ *
+ * @since [*next-version*]
+ */
+interface ContainerExceptionInterface extends ContainerAwareInterface
```

Comment:
Consider extending [`ContainerExceptionInterface`](https://github.com/php-fig/container/blob/master/src/ContainerExceptionInterface.php) from PSR-11, similar to how [`NotFoundExceptionInterface`](https://github.com/Dhii/data-container-interface/pull/1/files#diff-c35d7c858db72c95ecf92aba2df2199e) in this package extends the PSR-11 counterpart.